### PR TITLE
Add tests for design.Primitive.IsCompatible()

### PR DESCRIPTION
**Pull request by tchssk**
#1423

### CODE REVIEW

**Comment by raphael** [comment]
Great test! Do we need the line https://github.com/goadesign/goa/pull/1579/files#diff-aeba054339fa031f9755df256611bee8R136 ?
Also the linter is complaining about the declarations (https://travis-ci.org/goadesign/goa/builds/351249017#L623-L627), one way to address that could be to add the file to the top level `.lint_exclude` file. Thank you as always!

File: design/types_test.go [hunk 1]
```diff
@@ -109,3 +109,148 @@ func TestIsPrimitive(t *testing.T) {
...
+		f64  float64 = 20.2
+		s    string  = "string"
+		bs   []byte  = []byte("bytes")
+	)
+	cases := map[string]struct {
+		p        Primitive
+		values   []interface{}
+		expected bool
+	}{
+		//			values:   []interface{}{b, i, i8, i16, i32, ui, ui8, ui16, ui32, i64, ui64, f32, f64, s, bs},
```
Comment (tchssk) [inline design/types_test.go]: Oops, it's my temporary comment. https://github.com/goadesign/goa/pull/1579/commits/f094c7660955f602c7dc34ba563b1fe49b25af3f

**Comment by tchssk** [comment]
https://github.com/goadesign/goa/pull/1579/commits/98c3595f2bc09649aebca7e9487f46922ae23a81 uses type conversions. It resolved lint errors.

**Comment by raphael** [comment]
Great, thank you!

**Pull request merged**